### ISSUE TITLE
Colors Broken on `next`

### ISSUE THREAD

**Issue by bhilburn** [bhilburn/powerlevel9k#702]
I recently merged #684, a great enhancement PR from @Artistan for color definitions, into `next`.

I am only just now discovering that it breaks some colors in iTerm2 in interesting ways. It seems that any colorcode beyond the standard offered by ZSH (black, green, blue, red, magenta, cyan, white, yellow), when used in a background, is replaced with the foreground color. 

For example, using the `grey50` code instead of the `244` octal draws:
<img width="185" alt="screen shot 2017-12-14 at 16 22 45" src="https://user-images.githubusercontent.com/569571/34015002-17082a92-e0eb-11e7-98fd-d253a654d30c.png">

Using the standard colors appears to break in a way I honestly don't totally understand:
<img width="191" alt="screen shot 2017-12-14 at 16 23 40" src="https://user-images.githubusercontent.com/569571/34015040-399ddca0-e0eb-11e7-99e6-73b35480f0b7.png">

That example is from the `history` segment. If I revert it from `grey50` back to `244`, it works fine. There seem to be a select few standard colors that also work fine, like yellow and red.

@Artistan - Do you have any thoughts as to what might be happening, here?

**Comment by Artistan** [comment]
Interesting. Let me take a look.

**Comment by Artistan** [comment]
Can you post your config?

**Comment by Artistan** [comment]
One thing I have noticed in iterm2 is that it uses the colors defined in iterm to override your settings. This is part of the reason that originally did this pull request. Then you can adjust colors on the fly with your terminal settings (based on color names) rather than adjusting the config for powerlevel9k. It could be that your current color settings are odd in iterm.  The master version of powerlevel9k does not work with iterm2 color settings, and actually breaks when you are in a dark theme (iterm2) due to some of the colors being numbers, and some of them being words that can be changed with iterm2.  The #682 PR fixes that issue, but may have cause other issues???

![image](https://user-images.githubusercontent.com/801349/34045732-d66b67e4-e16f-11e7-98a5-def91e86fcca.png)

please post your config so I can check it out.

**Comment by Artistan** [comment]
light vs dark seems to be part of my "issue"
if you change your colors in iterm it will adjust the output

## "Light Background"
![image](https://user-images.githubusercontent.com/801349/34047335-8d3830b0-e175-11e7-87fc-fc30c15a7bae.png)
## "Dark Background"
![image](https://user-images.githubusercontent.com/801349/34047345-982f3ed2-e175-11e7-8ee7-6d65e688b01a.png)
## "Dark Background" adjusted
with fonts adjusted so it is not black text on black background
![image](https://user-images.githubusercontent.com/801349/34047577-4fa30a44-e176-11e7-984f-43365d7ef0d1.png)
## "Dark Background" with "light" powerline
![image](https://user-images.githubusercontent.com/801349/34047640-84e3a808-e176-11e7-8f24-963b120ed177.png)
.## "Dark Background" with "light" powerline
with fonts adjusted so it is not black text on black background
![image](https://user-images.githubusercontent.com/801349/34047697-a9a3afe4-e176-11e7-9b04-9ff8beeef5a0.png)

**Comment by Artistan** [comment]
please save this as colors.sh somewhere and run `zsh colors.zsh`
paste the output here to confirm my findings.
```

function getColor() {
    test="%K{$1}"
    result=$(print -P "$test")
    printf "%q" $result
    if [[ -n $result ]]; then
        echo "$result - non zero"
    else
        echo "$result non est"
    fi
}

getColor white
getColor red
getColor black
getColor abcde
getColor fghij
getColor lkmno


```

**Comment by Artistan** [comment]
getting there...
![image](https://user-images.githubusercontent.com/801349/34060806-fe616c22-e1aa-11e7-8168-8056011152f4.png)

**Comment by bhilburn** [comment]
Hey!

Okay, so you can see my `zshrc` here: 
[https://github.com/bhilburn/dotfiles/blob/master/zshrc-macos](https://github.com/bhilburn/dotfiles/blob/master/zshrc-macos)

I ran that command under both `master` and `next` and got the same results:

`master`:
<img width="199" alt="screen shot 2017-12-15 at 16 19 52" src="https://user-images.githubusercontent.com/569571/34060957-01cd5458-e1b4-11e7-8fa2-f0363af8c21c.png">

`next`:
<img width="192" alt="screen shot 2017-12-15 at 16 20 10" src="https://user-images.githubusercontent.com/569571/34060960-0620c288-e1b4-11e7-82af-79214f846dfe.png">

**Comment by Artistan** [comment]
yep, that code was  testing the terminal more than the theme.

**Comment by Artistan** [comment]
part of the issue here is the fact that there are no tests for colors :)

**Comment by Artistan** [comment]
The pull request I submitted will check the terminal to see if named color is supported, then return the color code in the case that the named color is not supported.

**Comment by Artistan** [comment]
updated again with a patch for the right side segment separators.

**Comment by bhilburn** [comment]
Fixed by merge of #703!
